### PR TITLE
Add direct column solves

### PR DESCRIPTION
**Pull request by lcw**

### CODE REVIEW

**Comment by lcw** [comment]
bors try

File: slurmci-test.toml [hunk 1]
```diff
@@ -31,11 +33,13 @@ cpu_gpu = [
   { file = "test/DGmethods_old/sphere/advection_sphere_ssp33.jl", n = 2, args = [] },
   { file = "test/DGmethods_old/sphere/advection_sphere_ssp34.jl", n = 2, args = [] },
   { file = "test/LinearSolvers/poisson.jl", n = 2, args = [] },
+  { file = "test/LinearSolvers/columnwiselu.jl", n = 1, args = []),
```
Comment (simonbyrne) [inline slurmci-test.toml]:
```suggestion
  { file = "test/LinearSolvers/columnwiselu.jl", n = 1, args = []},
```

**Comment by lcw** [comment]
bors try

File: slurmci-test.toml (hunk 1 above)
Comment (lcw) [reply slurmci-test.toml]: thanks

File: test/LinearSolvers/bandedsystem.jl [hunk 2]
```diff
@@ -0,0 +1,224 @@
+using CLIMA: haspkg
```
Comment (simonbyrne) [inline test/LinearSolvers/bandedsystem.jl]: the tests are looking for this in `DGmethods`

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by lcw** [comment]
bors try

File: test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion_1dimex.jl [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+               CentralGradPenalty(),
+               direction=EveryDirection())
+
+  hdg = DGModel(model,
+                grid,
+                Rusanov(),
+                CentralNumericalFluxDiffusive(),
+                CentralGradPenalty(),
+                auxstate=dg.auxstate,
+                direction=HorizontalDirection())
```
Comment (jkozdon) [inline test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion_1dimex.jl]: Remove these lines?

Comment (lcw) [reply test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion_1dimex.jl]: sure

**Comment by lcw** [comment]
bors try-

File: src/LinearSolvers/ColumnwiseLUSolver.jl [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+const LS = LinearSolvers
+using ..MPIStateArrays
+using LinearAlgebra
+using GPUifyLoops
+
+include("ColumnwiseLUSolver_kernels.jl")
+
+abstract type AbstractColumnLUSolver <: AbstractLinearSolver  end
+
+struct ManyColumnLU <: AbstractColumnLUSolver  end
```
Comment (simonbyrne) [inline src/LinearSolvers/ColumnwiseLUSolver.jl]: What is the difference between `ManyColumnLU` vs `SingleColumnLU`?

**Comment by lcw** [comment]
For any hope at performance we need the julia change https://github.com/JuliaLang/julia/pull/33678.

File: src/LinearSolvers/ColumnwiseLUSolver_kernels.jl [hunk 5]
```diff
@@ -0,0 +1,411 @@
+using Requires
+@init @require CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
+  using .CUDAnative
+end
+using StaticArrays
+
+"""
+    band_lu_knl!(A, Val(Nq), Val(Nqj), Val(nstate), Val(nvertelem),
```
Comment (simonbyrne) [inline src/LinearSolvers/ColumnwiseLUSolver_kernels.jl]: Signature doesn't match method below (`Val(Nq)` has been split it looks like)

File: slurmci-test.toml [hunk 6]
```diff
@@ -3,45 +3,49 @@
...
+  { file = "test/DGmethods_old/Euler/isentropic_vortex_standalone_integral.jl", slurmargs = ["--ntasks=3"], args = [] },
+  { file = "test/DGmethods_old/Euler/isentropic_vortex_standalone_source.jl", slurmargs = ["--ntasks=3"], args = [] },
+  { file = "test/DGmethods_old/compressible_Navier_Stokes/mms_bc.jl", slurmargs = ["--ntasks=3"], args = [] },
+  { file = "test/DGmethods_old/conservation/sphere.jl", slurmargs = ["--ntasks=3"], args = [] },
+  { file = "test/DGmethods_old/sphere/advection_sphere_lsrk.jl", slurmargs = ["--ntasks=2"], args = [] },
+  { file = "test/DGmethods_old/sphere/advection_sphere_ssp33.jl", slurmargs = ["--ntasks=2"], args = [] },
+  { file = "test/DGmethods_old/sphere/advection_sphere_ssp34.jl", slurmargs = ["--ntasks=2"], args = [] },
+  { file = "test/LinearSolvers/poisson.jl", slurmargs = ["--ntasks=2"], args = [] },
+  { file = "test/LinearSolvers/columnwiselu.jl", slurmargs = ["--ntasks=1"], args = []},
+  { file = "test/LinearSolvers/bandedsystem.jl", slurmargs = ["--ntasks=3"], args = [] },
```
Comment (simonbyrne) [inline slurmci-test.toml]: CPU is timing out:
```suggestion
  { file = "test/LinearSolvers/bandedsystem.jl", slurmargs = ["--ntasks=3 --time=02:00:00"], args = [] },
```
or use more tasks?

GPU test is failing.

File: src/LinearSolvers/ColumnwiseLUSolver.jl (hunk 4 above)
Comment (lcw) [reply src/LinearSolvers/ColumnwiseLUSolver.jl]: Single column assumes the system is the same for each column.  I have pushed docs that discuss this.

File: src/LinearSolvers/ColumnwiseLUSolver_kernels.jl (hunk 5 above)
Comment (lcw) [reply src/LinearSolvers/ColumnwiseLUSolver_kernels.jl]: Thanks, fixed.

File: slurmci-test.toml (hunk 6 above)
Comment (jkozdon) [reply slurmci-test.toml]: Currently working on tracking down the GPU bug

**Comment by lcw** [comment]
@simonbyrne I am not sure why the build is failing.

**Comment by simonbyrne** [comment]
I'll look into it

**Comment by lcw** [comment]
Initial timings of this work can be found in this gist <https://gist.github.com/lcw/897553496ee21b2149fd47548dbb09e9>.

**Pull request merged**